### ISSUE TITLE
scalikejdbc-mapper-generator (scalikejdbcGenForceAll) improvement request

### ISSUE THREAD

**Issue by kpopovic** [scalikejdbc/scalikejdbc#509]
Hello,

I would like to enlist several improvements regarding scalikejdbc-mapper-generator.

- improvement 1: scalikejdbcGenForceAll - it would be nice if could also read views and generate case classes,  not just tables. I know that you we can't write to view, but generated case classes from view are pretty handy.

- improvement 2: Add to the scalikejdbcGenForceAll parameter 
"suffix" so that all DB entity classes can be clearly distinguished, e.g. scalikejdbcGenForceAll suffix=DAO --> example: generates AddressDAO (from address DB table), AuthenticationDAO (from authentication DB table),...etc...

- improvement 3: skip generation of unit tests if some parameter is set. Maybe this exists already ?
e.g. scalikejdbcGenForceAll suffix=DAO getTest=false

ps. I read http://scalikejdbc.org/documentation/setup.html but I couldn't find how to skip unit test generating.

**Comment by seratch** [comment]
Your suggestions look fine. :+1: 

To be honest, working on mapper-generator is not a high priority task for me. It would be appreciated if anyone (yes, including you) can work on it.

**Comment by xuwei-k** [comment]
> improvement 2: Add to the scalikejdbcGenForceAll parameter 
> "suffix" so that all DB entity classes can be clearly distinguished

You can setting by `tableNameToClassName: String => String`

- https://github.com/scalikejdbc/scalikejdbc/blob/2.3.5/scalikejdbc-mapper-generator/src/main/scala/scalikejdbc/mapper/SbtPlugin.scala#L28
- https://github.com/scalikejdbc/scalikejdbc/blob/2.3.5/scalikejdbc-mapper-generator/src/main/scala/scalikejdbc/mapper/SbtKeys.scala#L14

**Comment by kpopovic** [comment]
Thank you for the quick answer. I will try to contribute to the improvement.

@xuwei-k : Did you want to say that this feature already exists or I will have to do implementation update  in "tableNameToClassName: String => String" ?

**Comment by kpopovic** [comment]
Hello,

I think I found it, but not sure.

https://github.com/scalikejdbc/scalikejdbc/blob/master/scalikejdbc-mapper-generator-core/src/main/scala/scalikejdbc/mapper/CodeGenerator.scala

constructor: class CodeGenerator(table: Table, specifiedClassName: Option[String] = None)

maybe to add another parameter: suffix:Option[String]
class CodeGenerator(table: Table, specifiedClassName: Option[String] = None, suffix:Option[String] = None)

This param could be read from config properties file like other parameters which are already read.

**Comment by xuwei-k** [comment]
> improvement 3: skip generation of unit tests if some parameter is set. 

https://github.com/scalikejdbc/scalikejdbc/blob/2.3.5/scalikejdbc-mapper-generator-core/src/main/scala/scalikejdbc/mapper/CodeGenerator.scala#L994

I think already exists. Scalikejdbc does not generate tests if specify invalid test template name in `project/scalikejdbc.properties` like following.

```
generator.packageName=models
generator.lineBreak=LF
generator.template=queryDsl
generator.testTemplate=some_invalid_test_template_name
```

**Comment by kpopovic** [comment]
Hello,

I have set generator.testTemplate=some_invalid_test_template_name and I can confirm that unit tests are not generated.

**Comment by kpopovic** [comment]
:+1:

**Comment by seratch** [comment]
Fixed by #510
